Lib/asyncio/locks.py

```diff
@@ -105,19 +108,16 @@ async def acquire(self):
         fut = self._get_loop().create_future()
         self._waiters.append(fut)
 
-        # Finally block should be called before the CancelledError
-        # handling as we don't want CancelledError to call
-        # _wake_up_first() and attempt to wake up itself.
```

Comment:
(above comment was removed because it no longer relevant, IMO)
Yes, that is correct.  Apart from catching not just CancelledError, it also avoids an unnecessariy complex nesting of try/catch, probably a historical accident from the time when `self._waiters.remove()` was moved from the `_wake_up_first()` method.

I know that the current code base only allows for CancelledError to be raised but I have been working on a package allowing other exceptions to be sent to tasks, and so I have a personal interest in them being handled properly  :)